src/main/java/uk/gov/hmcts/darts/common/repository/CaseRepository.java

```diff
@@ -83,13 +84,11 @@ List<CourtCaseEntity> findCasesNeedingCaseDocumentForRetentionDateGeneration(Off
                                                                                  Pageable pageable);
 
     @Query(value = """
-        select cc.* from darts.darts.court_case cc
-        join darts.darts.case_retention cr
-        on cr.cas_id = cc.cas_id and cr.current_state = 'COMPLETE'
-        where cc.is_data_anonymised = false
-        and cr.retain_until_ts < :maxRetentionDate
-        limit :limit
-        """,
-        nativeQuery = true)
-    List<CourtCaseEntity> findCasesToBeAnonymized(OffsetDateTime maxRetentionDate, long limit);
+        select cc from CourtCaseEntity cc
+        join CaseRetentionEntity cr
+        on cr.courtCase.id = cc.id and cr.currentState = 'COMPLETE'
+        where cc.isDataAnonymised = false
+        and cr.retainUntil < :maxRetentionDate
+        """)
+    List<CourtCaseEntity> findCasesToBeAnonymized(OffsetDateTime maxRetentionDate, Limit limit);
```

Comment:
`Limit` is not used anywhere in the current code effectively and should be replaced with Pageable for limit and pagination functionality

Reply:
We use limit instead of Pageable here because it is more performant as Pageable gets the total counts of entries that match the criteria where Limit only grabs the limit so less SQL